src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -116,8 +117,35 @@ public ScanResults scan(ScanRequest scanRequest) {
             report.log();
             return getEmptyScanResults();
         }
-    }
+    } 
 
+    @Override
+    public ScanResults scanCli(ScanRequest request, String scanType, File... files) {
+        ScanResults scanResults = null;
+        try {
+            switch (scanType) {
+                case "Scan-git-clone":
+                    scanResults = cxFullScan(request);
+                    break;
+                case "cxFullScan":
+                    scanResults = cxFullScan(request, files[0].getPath());
+                    break;
+                case "cxParse":
+                    cxParseResults(request, files[0]);
+                    break;
+                case "cxBatch":
+                    cxBatch(request);
+                    break;
+                default:
+                    log.warn("SastScanner does not support scanType of {}, ignoring.");
+                    break;
+            }
```

Comment:
can this method go one level up? it should do the same thing in the different scanners
in this case each scanner should implement only the different scan types 

let's consider it in the next task on this area

Reply:
this is part of the decupling, this way if some scanner adds a new type, it will not force a change in the others